### PR TITLE
Fixing ARX na > nb

### LINKED ISSUE #75: arx() estimator issues
Apparent issues with several of the arx() estimators used in your identification_arx.ipynb example with slightly modified AR and ARMA transfer functions with the following solvers: (I assume they all should handle the MA part also given nb is specified? I did not find an armax version)
```julia
Gls      = arx(d,na,nb)                      # Regular least-squares estimation
Gtls     = arx(d,na,nb, estimator=tls)       # Total least-squares estimation
Gwtls    = arx(d,na,nb, estimator=wtls_estimator(y,na,nb)) # Weighted Total least-squares estimation
Gplr, Gn = plr(d,na,nb,nc, initial_order=20) # Pseudo-linear regression
Gn4sid = tf(n4sid(d, :auto; verbose=false).sys) #
```

1) Tested  pure AR with two imaginary poles  on
```julia
     0.8
-------------
1.0z^2 + 0.25
```

A small error is added to simulated data
```julia
y  = sim(G,u)

eps = 0.000001
yn = y + eps * randn(size(y));
```
Testing various specifications of system order (denominator and numerator). I am a bit uncertain if n is the number of coefficients here or the order?

With na=2,nb=2,nc=1 (?) they all gave reasonably similar and ok results..
```julia
Gwtls = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
     -1.830761022395132e-8z + 0.8000000364406703
-----------------------------------------------------
1.0z^2 + 1.1336180899546022e-7z + 0.24999999923772648

Gn4sid = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
1.0482460250016439e-7z^2 - 2.4016742360023076e-8z + 0.8000000313485508
----------------------------------------------------------------------
          1.0z^2 + 7.743629093504012e-8z + 0.250000013686116

```
With na=2, nb=1 (should suffice for the numerator)  the arx versions start failing bad, while n4sid still does well

```julia
Gls = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
              -0.008425068831949092
-------------------------------------------------
1.0z^2 - 0.0149157451820707z + 0.2485242562797274


Gtls = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
               -0.14710343635360346
--------------------------------------------------
1.0z^2 - 0.10034853554694595z + 1.0435201347058614


Gplr = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
               -0.00843907141354728
---------------------------------------------------
1.0z^2 - 0.01502858249704049z + 0.25577236172028245


Gwtls = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
               -0.06823555574414619
--------------------------------------------------
1.0z^2 - 0.09093792215356927z + 0.9887132202347376


Gn4sid = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
-7.035087559097725e-8z^2 + 1.8586494326913326e-8z + 0.7999999720915703
----------------------------------------------------------------------
         1.0z^2 - 4.118840077338426e-8z + 0.2500000350249592
```

With  na=nb=4 (too high?) , the arx and plr also goes bad (large errs in both numerator and denom) Again n4sid good.
```julia
Gls = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
     4.498197651604916e-8z^3 + 0.7999999908664831z^2 + 0.003570318828339773z - 0.11531636330533376     
-------------------------------------------------------------------------------------------------------
1.0z^4 + 0.0044629928399430585z^3 + 0.10585461073330074z^2 + 0.001115779335659886z - 0.0360363583918387

Gtls = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
    2.7838920503841905e-8z^3 + 0.8000000588632808z^2 + 0.891535197737989z + 0.6889073529493018
--------------------------------------------------------------------------------------------------     
1.0z^4 + 1.1144190526902322z^3 + 1.111134362289782z^2 + 0.27860493890406324z + 0.21528368611087234     

Gplr = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
      4.420389876275559e-8z^3 + 0.8000000013003751z^2 - 0.003497016602469035z - 0.1134473493326818
---------------------------------------------------------------------------------------------------------
1.0z^4 - 0.0043711788328240336z^3 + 0.10819087189424594z^2 - 0.0010927728758850003z - 0.03545228883099305

Gwtls = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
  1.5679542424341932e-8z^3 + 0.8000000093945829z^2 + 0.038695551344337255z + 0.45693930399129623
--------------------------------------------------------------------------------------------------
1.0z^4 + 0.04836949844579501z^3 + 0.821174204714606z^2 + 0.01209246301841781z + 0.1427935848734602

Gn4sid = TransferFunction{Discrete{Float64}, ControlSystems.SisoRational{Float64}}
1.8043888767223154e-8z^2 + 6.290335821529143e-8z + 0.8000000000517333
---------------------------------------------------------------------
        1.0z^2 + 7.375779662033288e-8z + 0.25000005919139123
```
As can be seen the n4sid has typical coefficient errors  in the e-8 range, while the others are up to e-1
Per Ove Husøy

### PR DESCRIPTION
**Pull request by mapi1**
This fixes #75 and adds some more tests. Also some minor adjustments in the docs.

### CODE REVIEW

File: src/arx.jl [hunk 1]
```diff
@@ -119,8 +119,8 @@ end
...
-- `nb` and `na` are the length of the numerator and denominator polynomials. Input delay can be added via `inputdelay = d`, which corresponds to an additional delay of `z^-1`.  An `inputdelay = 0` results in a direct term.  `λ > 0` can be provided for L₂ regularization. `estimator` defaults to \\ (least squares), alternatives are `estimator = tls` for total least-squares estimation. `arx(Δt,yn,u,na,nb, estimator=wtls_estimator(y,na,nb)` is potentially more robust in the presence of heavy measurement noise.
-The number of free parameters is `na+nb`
+
+- `nb` and `na` are the number of coefficients of the numerator and denominator polynomials. Input delay can be added via `inputdelay = d`, which corresponds to an additional delay of `z^-1`.  An `inputdelay = 0` results in a direct term. The highest order of the B polynomial is given by `nb + inputdelay - 1`.  `λ > 0` can be provided for L₂ regularization. `estimator` defaults to \\ (least squares), alternatives are `estimator = tls` for total least-squares estimation. `arx(Δt,yn,u,na,nb, estimator=wtls_estimator(y,na,nb)` is potentially more robust in the presence of heavy measurement noise. The number of free parameters is `na+nb`
```
Comment (baggepinnen) [inline src/arx.jl:123]:
```suggestion
- `nb` and `na` are the number of coefficients of the numerator and denominator polynomials.
Input delay can be added via `inputdelay = d`, which corresponds to an additional delay of `z^-d`.
An `inputdelay = 0` results in a direct term.
The highest order of the B polynomial is given by `nb + inputdelay - 1`.  `λ > 0` can be provided for L₂ regularization.
`estimator` defaults to \\ (least squares), alternatives are `estimator = tls` for total least-squares estimation. 
`arx(Δt,yn,u,na,nb, estimator=wtls_estimator(y,na,nb)` is potentially more robust in the presence of
heavy measurement noise. The number of free parameters is `na+nb` 
```

Change "additional delay of `z^-1`" to "additional delay of `z^-d`"

File: test/runtests.jl [hunk 2]
```diff
@@ -512,6 +536,16 @@ freqresptest(G, model, tol) = freqresptest(G, model) < tol
...
         
 
+        ## Test na > nb
+        na, nb, inputdelay = 2, 1, 2
+        G1   = tf(0.8, [1,0.5,0.25], 1)
+        u = randn(N)
+        y = lsim(G1, u, t)[1][:]
+        d = iddata(y, u, 1)
+        Gest = arx(d, na, nb, inputdelay = inputdelay)
+        @test G1 ≈ minreal(Gest)
```
Comment (baggepinnen) [inline test/runtests.jl:546]: Why is `minreal` required here? Shouldn't the parameters `na, nb, inputdelay = 2, 1, 2` result in a transfer function of the same order as `G1`?

Comment (mapi1) [reply test/runtests.jl:546]: Yes, this minreal() is only a copy-paste artifact and the test should work without it just fine.

Comment (baggepinnen) [inline test/runtests.jl:546]:
```suggestion
        @test G1 ≈ Gest
```

File: test/runtests.jl [hunk 3]
```diff
@@ -470,8 +484,18 @@ freqresptest(G, model, tol) = freqresptest(G, model) < tol
...
 
-        ## MISO nb1 != nb2
+        ## Test na > nb
+        na, nb = 2, 1
+        G1  = tf([0.8, 0], [1,0.5,0.25], 1)
+        u = randn(N)
+        y = lsim(G1, u, t)[1][:]
+        d = iddata(y, u, 1)
+        Gest = arx(d, na, nb)
+        @test G1 ≈ minreal(Gest)
```
Comment (baggepinnen) [inline test/runtests.jl:494]:
```suggestion
        @test G1 ≈ Gest
```

**Comment by baggepinnen** [comment]
Excellent, thanks for your help!

**Pull request merged**